packages/services/f-analytics/src/services/_tests/analytics.service.test.js

```diff
@@ -16,55 +17,256 @@ describe('Analytic Service ::', () => {
+        }
+        windowCopy = { ...global.window };
         const innerWidthSpy = jest.fn().mockReturnValue(winWidth);
         const innerHeightSpy = jest.fn().mockReturnValue(winHeight);
-        windowSpy = jest.spyOn(global, 'window', 'get');
-        windowSpy.mockImplementation(() => ({
+        windowsPushSpy = jest.fn();
+        const windowMock = ({
             ...windowCopy,
-            dataLayer: {
-                push: windowsPushSpy
-            },
+            dataLayer: dataLayerPresent ? { push: windowsPushSpy } : undefined,
             innerWidth: innerWidthSpy(),
             innerHeight: innerHeightSpy()
-        }));
+        });
+        windowSpy = jest.spyOn(global, 'window', 'get');
+        windowSpy.mockImplementation(() => windowMock);
     };
 
-    beforeEach(() => {
-        // Arrange - store
+    const mockStore = () => {
         store = createStore({
+            state: { ...defaultState },
             actions: analyticModule.actions,
             mutations: analyticModule.mutations
         });
-        storeDispatchSpy = jest.spyOn(store, 'dispatch');
+
+        // Hyjack and replicate updating the State thru dispatch calls
+        storeDispatchSpy = jest.fn((action, payload) => {
+            if (action === `${options.namespace}/updateEvents`) {
+                store.state[`${options.namespace}`].events = [...store.state[`${options.namespace}`].events, payload];
+            } else if (action === `${options.namespace}/clearEvents`) {
+                store.state[`${options.namespace}`].events = [];
+            } else if (action === `${options.namespace}/updatePlatformData`) {
+                store.state[`${options.namespace}`].platformData = { ...store.state[`${options.namespace}`].platformData, ...payload };
+            }
+        });
+        store.dispatch = storeDispatchSpy;
+
+        registerStoreModuleSpy = jest.fn();
+        store.registerModule = registerStoreModuleSpy;
+    };
+
+    beforeEach(() => {
+        req = jest.fn();
+
+        // Arrange - store
+        mockStore();
+
         // Arrange - cookies
         get = jest.fn();
         when(get).calledWith('je-auser').mockReturnValue(auserId);
+        when(get).calledWith('je-user_percentage').mockReturnValue('35');
         Cookies.mockImplementation(() => ({ get }));
+
         // Arrange - window state
         mockWindow();
+
         // Arrange - sut
-        service = new AnalyticService(store, options);
+        service = new AnalyticService(store, req, options);
     });
 
     afterEach(() => {
+        store = undefined;
+        service = undefined;
+        windowSpy.mockRestore();
+        storeDispatchSpy.mockRestore();
         jest.clearAllMocks();
     });
 
-    it('should instantiate a new instance', () => {
-        // Act
-        const instance = () => new AnalyticService(store, options);
+    describe('When creating a new instance', () => {
+        it('should instantiate a new instance', () => {
```

Comment:
Maybe this should read..
`should not throw error when instance is created with valid option types` ?